### PR TITLE
Move function calls with static arguments out of loops

### PR DESCRIPTION
**Pull request by mitchute**
See issue #5832 for an example. Review the code and cleanup these issues.

### Work Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [x] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [x] At least one of the following appropriate labels must be added to this PR to be consumed into the changelog:
   - Defect: This pull request repairs a github defect issue.  The github issue should be referenced in the PR description
   - Refactoring: This pull request includes code changes that don't change the functionality of the program, just perform refactoring
   - NewFeature: This pull request includes code to add a new feature to EnergyPlus
   - Performance: This pull request includes code changes that are directed at improving the runtime performance of EnergyPlus
   - DoNoPublish: This pull request includes changes that shouldn't be included in the changelog

### Review Checklist
This will not be exhaustively relevant to every PR.
 - [x] Code style (parentheses padding, variable names)
 - [x] Functional code review (it has to work!)
 - [ ] CI status: all green or justified
 - [ ] Performance: CI Linux results include performance check -- verify this

### CODE REVIEW

**Comment by rraustad** [comment]
36% improvement. Outstanding.

**Comment by Myoldmopar** [comment]
> 36% improvement. Outstanding.

Please hold, a bunch of those performance files failed, so they are probably showing up as zeroes.  I'm definitely hopeful for good improvement from @mitchute but that might be a bit too high.

**Comment by JasonGlazer** [comment]
@mitchute thanks for taking this one on. I will be happy to review this. Hopefully most of these CI issue will be easy to fix.

**Comment by mitchute** [comment]
I'm hopeful this next round of CI results will give some insight into the savings.

**Comment by rraustad** [comment]
This is the type of err change that says something is very different.
I forgot the specific file name now but I think it was DOAToFanCoilInlet

        ************* SizeHVACSingleDuct: Potential issue with equipment sizing for AirTerminal:SingleDuct:VAV:NoReheat = "SPACE1-1 DOAS AIR TERMINAL".
        **   ~~~   ** User-Specified Minimum Cooling Air Flow Fraction of 1.00000 [m3/s]
    -   **   ~~~   ** differs from Design Size Minimum Cooling Air Flow Fraction of 0.72766 [m3/s]
    +   **   ~~~   ** differs from Design Size Minimum Cooling Air Flow Fraction of 0.00000 [m3/s]

**Comment by rraustad** [comment]
Pardon me for interrupting. I merged develop to see if that eased the diff's.

**Comment by mitchute** [comment]
Thanks for merging that in @rraustad.

File: src/EnergyPlus/SolarShading.cc
Comment (rraustad) [inline src/EnergyPlus/SolarShading.cc]: any requirement for spacing here? e.g., SinPhi[ IPhi ]

File: src/EnergyPlus/TarcogShading.cc
Comment (rraustad) [inline src/EnergyPlus/TarcogShading.cc]: I'm no expert but I see a difference in initialization syntax, e.g., x( y ) versus x = y.

File: src/EnergyPlus/SolarShading.cc
Comment (mitchute) [reply src/EnergyPlus/SolarShading.cc]: Probably should add them, but I'd guess that @Myoldmopar has the final say. As with everything, there's a variety with and without throughout the code.

File: src/EnergyPlus/TarcogShading.cc
Comment (mitchute) [reply src/EnergyPlus/TarcogShading.cc]: I've been trying to keep it consistent by going with ```Real64 cos_foo( std::cos( foo ) );```. I just missed this one. Again, any preference, @Myoldmopar ?

File: src/EnergyPlus/SolarShading.cc
Comment (Myoldmopar) [reply src/EnergyPlus/SolarShading.cc]: I prefer appropriate padding.  The convention EnergyPlus adopted from the translation is to pad everything.  Perhaps we can make a huge push to clear out PRs after the release and make a big move to finalize our style and get a style checker to verify pushed code passes.

File: src/EnergyPlus/TarcogShading.cc
Comment (Myoldmopar) [reply src/EnergyPlus/TarcogShading.cc]: This is another one that I differ from the adopted convention.  I prefer the equals, and there are subtle differences, but not for these specific calls.  

These types of things alone don't always hold up a PR from going in, as our style guide doesn't even match the code, which results in some spots of the code adopting new patterns.

**Comment by rraustad** [comment]
My first thought is that there should be no diff's. I have, at times, thought that a compilation of different code may show minor diff's (E-15), but major diff's should not be present here. Do you agree?

File: src/EnergyPlus/AirflowNetworkSolver.cc
Comment (JasonGlazer) [inline src/EnergyPlus/AirflowNetworkSolver.cc]: This line should not be moved out of the while loop. It depends on FT which depends on FTT which changes each time through the loop.

File: src/EnergyPlus/EcoRoofManager.cc
Comment (JasonGlazer) [inline src/EnergyPlus/EcoRoofManager.cc]: I don't think pow_3_SoilTK or pow_4_SoilTK can be precalculated outside of the loop because later on it is computed again within the loop

File: src/EnergyPlus/EcoRoofManager.cc
Comment (JasonGlazer) [inline src/EnergyPlus/EcoRoofManager.cc]: I don't think pow_3_LeafTK or pow_4_LeafTK can be precalculated outside of the loop because later on LeafTK is computed again within the loop

File: src/EnergyPlus/FuelCellElectricGenerator.cc
Comment (JasonGlazer) [inline src/EnergyPlus/FuelCellElectricGenerator.cc]: Tkel and Tsho should be const

File: src/EnergyPlus/FuelCellElectricGenerator.cc
Comment (JasonGlazer) [inline src/EnergyPlus/FuelCellElectricGenerator.cc:2111]: Tsho and Tkel should be const

File: src/EnergyPlus/FuelCellElectricGenerator.cc
Comment (JasonGlazer) [inline src/EnergyPlus/FuelCellElectricGenerator.cc:2214]: Tsho and Tkel should be const

File: src/EnergyPlus/FuelCellElectricGenerator.cc
Comment (JasonGlazer) [inline src/EnergyPlus/FuelCellElectricGenerator.cc:2316]: Same as above.

File: src/EnergyPlus/FuelCellElectricGenerator.cc
Comment (JasonGlazer) [inline src/EnergyPlus/FuelCellElectricGenerator.cc:2422]: Same as above.

File: src/EnergyPlus/FuelCellElectricGenerator.cc
Comment (JasonGlazer) [inline src/EnergyPlus/FuelCellElectricGenerator.cc:2533]: Same.

File: src/EnergyPlus/FuelCellElectricGenerator.cc
Comment (JasonGlazer) [inline src/EnergyPlus/FuelCellElectricGenerator.cc:2645]: Same.

File: src/EnergyPlus/FuelCellElectricGenerator.cc
Comment (JasonGlazer) [inline src/EnergyPlus/FuelCellElectricGenerator.cc:2757]: Same

File: src/EnergyPlus/GroundHeatExchangers.cc
Comment (JasonGlazer) [inline src/EnergyPlus/GroundHeatExchangers.cc:519]: These changes are ok but since they are not in a loop they will probably not have much performance impact.

File: src/EnergyPlus/HeatBalanceManager.cc
Comment (JasonGlazer) [inline src/EnergyPlus/HeatBalanceManager.cc]: This line could also be moved to be precalculated since CosPhi[IPhi} is a set of constants

File: src/EnergyPlus/SolarReflectionManager.cc
Comment (JasonGlazer) [inline src/EnergyPlus/SolarReflectionManager.cc:1182]: Great catch!

File: src/EnergyPlus/SolarShading.cc
Comment (JasonGlazer) [inline src/EnergyPlus/SolarShading.cc:538]: Great idea since they are used all over this file.

File: src/EnergyPlus/SolarShading.cc
Comment (JasonGlazer) [inline src/EnergyPlus/SolarShading.cc]: Thanks!

File: src/EnergyPlus/WindowManager.cc
Comment (JasonGlazer) [inline src/EnergyPlus/WindowManager.cc]: SunAzimuth, SunAltitude, and SkyArea could all be precalculated outside of the SurfNum loop since N and M are constants. So could RelativeAzimuth and RelativeAltitude

File: src/EnergyPlus/WindowManager.cc
Comment (JasonGlazer) [inline src/EnergyPlus/WindowManager.cc]: Should also make Pi/180 a constant. I think we might have one for that already.

File: src/EnergyPlus/WindowManager.cc
Comment (JasonGlazer) [inline src/EnergyPlus/WindowManager.cc:8289]: Thank you. I hate if's like that.

**Review by JasonGlazer — CHANGES_REQUESTED** [review]
Overall I appreciate you going so much farther than just fixing the SkyDifSolarShading() issue with sin and cos. I did find a couple of issues that need to be fixed.

**Comment by mitchute** [comment]
@JasonGlazer @rraustad Thanks for your input. I think this last commit addresses your comments. Integration tests ran successfully on my local machine, but let's see what CI comes back with.

**Comment by JasonGlazer** [comment]
@mitchute let me know when the code changes are over and I will finish up the review.

**Comment by mitchute** [comment]
@JasonGlazer I think this is going to be it. We'll see what CI comes back with, but I'm not expecting any more changes.

File: src/EnergyPlus/AirflowNetworkSolver.cc
Comment (JasonGlazer) [reply src/EnergyPlus/AirflowNetworkSolver.cc]: Thanks for fixing.

File: src/EnergyPlus/EcoRoofManager.cc
Comment (JasonGlazer) [reply src/EnergyPlus/EcoRoofManager.cc]: Thanks for fixing

File: src/EnergyPlus/EcoRoofManager.cc
Comment (JasonGlazer) [reply src/EnergyPlus/EcoRoofManager.cc]: Thanks for fixing

File: src/EnergyPlus/FuelCellElectricGenerator.cc
Comment (JasonGlazer) [reply src/EnergyPlus/FuelCellElectricGenerator.cc]: OK

File: src/EnergyPlus/FuelCellElectricGenerator.cc
Comment (JasonGlazer) [reply src/EnergyPlus/FuelCellElectricGenerator.cc:2111]: ok

File: src/EnergyPlus/FuelCellElectricGenerator.cc
Comment (JasonGlazer) [reply src/EnergyPlus/FuelCellElectricGenerator.cc:2316]: ok

File: src/EnergyPlus/WindowManager.cc
Comment (JasonGlazer) [reply src/EnergyPlus/WindowManager.cc]: Looks good.

**Review by JasonGlazer — APPROVED** [review]
Looks good!

**Comment by mitchute** [comment]
@Myoldmopar I just merge develop in one more time to see if these tests get cleaned up. I'm not seeing any failures on my machines (Windows 7 and Mac), but let me know if you see anything else on this.

**Comment by JasonGlazer** [comment]
@mitchute it still seems to be having some small problems with the CI in a few places. Are you planning on looking into those more?

**Comment by mitchute** [comment]
@JasonGlazer All of those run fine locally on my machine, so I'm not sure what else to do at this point.

**Comment by mjwitte** [comment]
@JasonGlazer The failure you saw here for 5ZoneAirCooledWithDOASAirLoop on the linux integration coverage tests is also failing in develop.  So it's not an issue with this branch.

**Pull request merged**
